vllm/engine/output_processor/stop_checker.py

```diff
@@ -112,8 +112,8 @@ def check_stop_strings(
         for stop_str in stop:
             stop_string_len = len(stop_str)
             # Avoid searching already-searched text.
-            stop_index = output_text.find(stop_str,
-                                          -new_char_count - stop_string_len)
+            stop_index = output_text.find(stop_str, 
+                                          len(output_text) - new_char_count - stop_string_len + 1)
```

Comment:
This is safe since both new_char_count and stop_string_len must be >= 1.
```suggestion
            stop_index = output_text.find(stop_str, 1 - new_char_count - stop_string_len)
```

Reply:
Thanks for your suggestion. A new commit has been pushed which resolves DCO and code simplification.